docs/how-to/contribute-code.rst

```diff
@@ -24,12 +24,186 @@ dependencies:
     $ cd rubicon-objc
     $ python3 -m venv venv
     $ source venv/bin/activate
-    (venv) $ pip install --upgrade pip
-    (venv) $ pip install --upgrade setuptools
-    (venv) $ pip install tox
-    (venv) $ pip install -e .
+    (venv) $ python3 -m pip install -Ue ".[dev]"
 
-You can then run the full test suite:
+Rubicon uses a tool called `Pre-Commit <https://pre-commit.com>`__ to identify
+simple issues and standardize code formatting. It does this by installing a git
+hook that automatically runs a series of code linters prior to finalizing any
+git commit. To enable pre-commit, run:
+
+.. tabs::
+
+  .. group-tab:: macOS
+
+    .. code-block:: bash
+
+      (venv) $ pre-commit install
+      pre-commit installed at .git/hooks/pre-commit
```

Comment:
Definitely some copy-pasta here, since we don't need Linux/Windows instructions. I've deleted those blocks, and made the indentation consistent with the rest of the file (which, FWIW, is 4 characters in this file, and the other files I saw in a quick, but incomplete audit).

Reply:
Ah yep, looks like we use 4 spaces elsewhere. Not sure where I got the idea that reST requires exact visual indents - maybe that's only in some special cases.